velox/serializers/PrestoSerializer.cpp

```diff
@@ -949,6 +989,29 @@ void VectorStream::append(folly::Range<const IntervalDayTime*> values) {
   }
 }
 
+template <>
+inline void VectorStream::append(
+    folly::Range<const UnscaledLongDecimal*> values) {
+  constexpr int128_t kInt128SerializeMask = (static_cast<int128_t>(1) << 127);
+  for (auto& value : values) {
+    int128_t val = value.unscaledValue();
+    // Presto Java UnscaledDecimal128 representation uses signed magnitude
+    // representation. Only negative values differ in this representation.
+    // Due to this difference in representation, the UnscaledLongDecimal minimum
+    // value cannot be represented as a Presto UnscaledDecimal128 value. We
+    // throw an error in this case.
+    if (val < 0) {
+      if (value == std::numeric_limits<UnscaledLongDecimal>::min()) {
+        VELOX_FAIL(
+            "Cannot serialize '{}' as a Presto UnscaledDecimal128 value", val);
```

Comment:
We use this serializer for serde of spill data. Does this mean we won't be able to spill some values?

Reply:
@mbasmanova We should never touch this code path in practice. @karteekmurthys has another PR https://github.com/facebookincubator/velox/pull/2456 where the limits based on maximum precision (38) are enforced. The precision 38 limit is lower than the int128 limit. I should make this a VELOX_CHECK.

Reply:
To clarify, the LongDecimalType lower limits `-10^38 + 1 > - 2^127`.